### PR TITLE
Fix wrong location for inline asm blocks

### LINKED ISSUE #6505: wrong source location for asm node
The code 

```
contract C {
    function f() public pure returns (uint y) {
        uint a;
        assembly { a := 0x12345678 }
        uint z = a;
        y = z;
    }
}
```
has wrong asm location information. This is visible when, for example, running it with `solc --ast`, however `solc --ast-json` remains unaffected

### PR DESCRIPTION
**Pull request by Marenz**
fixes #6505

### CODE REVIEW

File: libsolidity/parsing/Parser.cpp [hunk 1]
```diff
@@ -1050,8 +1051,9 @@ ASTPointer<InlineAssembly> Parser::parseInlineAssembly(ASTPointer<ASTString> con
 	shared_ptr<yul::Block> block = asmParser.parse(m_scanner, true);
 	if (block == nullptr)
 		BOOST_THROW_EXCEPTION(FatalError());
-	nodeFactory.markEndPosition();
-	return nodeFactory.createNode<InlineAssembly>(_docString, block);
+
+	location.end = block->location.end;
```
Comment (chriseth) [inline libsolidity/parsing/Parser.cpp:1055]: Other code uses `nodeFactory.setEndPositionFromNode(block)`.

File: test/libsolidity/ParserTest.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	ASTPointer<SourceUnit> sourceUnit;
+	BOOST_REQUIRE_NO_THROW(sourceUnit = parser.parse(make_shared<Scanner>(_sourceCode)));
+	BOOST_CHECK(!!sourceUnit);
+
+	return sourceUnit;
+}
+
+} // end anonymous namespace
+
+BOOST_AUTO_TEST_SUITE(Parser)
```
Comment (chriseth) [inline test/libsolidity/ParserTest.cpp]: Why didn't you add this to SolidityParser.cpp?

File: test/libsolidity/ParserTest.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+			uint a;
+			assembly { a := 0x12345678 }
+			uint z = a;
+			y = z;
+		}
+	}
+	)", "");
+	auto sourceUnit = parseContract(sourceCode);
+
+	class CheckInlineAsmLocation : public ASTConstVisitor
```
Comment (chriseth) [inline test/libsolidity/ParserTest.cpp]:
```suggestion
	class CheckInlineAsmLocation: public ASTConstVisitor
```

File: test/libsolidity/ParserTest.cpp (hunk 2 above)
Comment (Marenz) [reply test/libsolidity/ParserTest.cpp]: I must have misunderstood your suggestion :)

File: libsolidity/parsing/Parser.cpp (hunk 1 above)
Comment (Marenz) [reply libsolidity/parsing/Parser.cpp:1055]: `block` is `yul::Block`, not `ASTPointer<ASTNode>`

**Comment by Marenz** [comment]
Updated

File: libsolidity/parsing/Parser.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/parsing/Parser.cpp:1055]: Ah, I see!

**Review by chriseth — APPROVED** [review]

**Pull request merged**